.github/actions/optimize-paths-filter/action.yml

```diff
@@ -0,0 +1,39 @@
+---
+name: Common paths filter for optimize
+
+description: Common filters to detect and group changes against a base branch
+
+outputs:
+  backend-changes:
+    description: Output whether any backend has changed
+    value: >-
+      ${{
+        github.event_name == 'push' ||
+        steps.filter-common.outputs.backend-change == 'true'
+      }}
+  frontend-changes:
+    description: Output whether any frontend has changed
+    value: >-
+      ${{
+        github.event_name == 'push' ||
+        steps.filter-common.outputs.frontend-change == 'true'
+      }}
+runs:
+  using: composite
+  steps:
+    - uses: dorny/paths-filter@v3
+      id: filter-common
+      with:
+        base: ${{ github.event.merge_group.base_ref || '' }}
+        ref: ${{ github.event.merge_group.head_ref || github.ref }}
+        filters: |
+          backend-change:
+            - optimize/backend/**
+            - optimize/upgrade/**
+            - optimize/util/**
+            - optimize/plugins/**
+
+          frontend-change:
+            - optimize/client/**
+            - optimize/c4/**
+            - optimize/backend/src/main/resources/localization/**
```

Comment:
in practice, does this mean the translation files? If so, I would argue that the BE workflows should also run. There are a number of ITs relating to making sure that various responses are localized/entities created correctly

Reply:
So, to backend changes we include the whole `optimize/backend` directory, so when there are changes to translation files, backend tests will run. For the frontend changes we only care for the localization files cause changing translations might break FE tests

Reply:
Ah yeah of course, that's obvious now. Got it - thanks 👍